--- orig	2016-12-12 01:32:49.000000000 +0100
+++ compat/precompose_utf8.c	2016-12-12 01:32:55.000000000 +0100
@@ -78,7 +78,13 @@
 		size_t namelen;
 		oldarg = argv[i];
 		if (has_non_ascii(oldarg, (size_t)-1, &namelen)) {
+#if NO_ICONV
+			newarg = malloc (namelen + 1);
+			memcpy (newarg, oldarg, namelen);
+			newarg[namelen] = 0;
+#else
 			newarg = reencode_string_iconv(oldarg, namelen, ic_precompose, NULL);
+#endif
 			if (newarg)
 				argv[i] = newarg;
 		}
