--- configure.orig	2017-01-02 13:00:41.000000000 +0100
+++ configure	2017-01-02 12:47:48.000000000 +0100
@@ -36,7 +36,7 @@
 parser = optparse.OptionParser()
 
 valid_os = ('win', 'mac', 'solaris', 'freebsd', 'openbsd', 'linux',
-            'android', 'aix')
+            'android', 'aix', 'iphoneos')
 valid_arch = ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'ppc', 'ppc64', 'x32',
               'x64', 'x86', 's390', 's390x')
 valid_arm_float_abi = ('soft', 'softfp', 'hard')
